Add Partially scanned PDFs preferences section and OCR shortcut (#642)
JabRef/user-documentation: en/advanced/OCR.md

OCRmyPDF must be installed on your system to use this feature.
{% endhint %}

1. Open JabRef and select the entry with the PDFs you want to OCR.

> 2. Scroll down the Main Tab of the Entry Editor till reach the Files and Links section.

Reviewer: ```suggestion
2. Scroll down the Main Tab of the Entry Editor till you reach the Files and Links section.
```

3. Right-click on the File and select "Perform OCR and embed text into new PDF file", or select the file and use the shortcut `Ctrl+Shift+R`. ![Perform OCR](../.gitbook/assets/perform-ocr.png)

* After performing OCR, JabRef creates a new PDF file with the OCR text embedded, and it will be linked to all the entries that have the old file linked to them. The original scanned PDF will remain unchanged.

    ![Original and OCRed files](../.gitbook/assets/original-and-ocred-files.png)

* Now you can select and search text in the new PDF file.

    ![Comparison between original and OCRed file](../.gitbook/assets/comparison-between-original-and-ocred-file.png)

## OCR Preferences

* The OCR preferences can be accessed via **File → Preferences → OCR**.

    ![OCR preferences](../.gitbook/assets/ocr-preferences.png)

> 

Reviewer: I have seen other apps structure OCR preferences via "Input" vs "Output".

> ### Partially scanned PDFs

Reviewer: Alternatives:
1. handling of pre-existing text
2. handling of original embedded text
3. handling of text input
4. text creation handling
5. handling of text creation
6. text flow handling
7. handling of source text

I am not quite sure what is best, but tend to favour 1 or 2. What do you think?
"Partially scanned PDFs" doesn't seem to be entirely correct. First of all we are not scanning. Secondly, overwriting is not handling it partially.

Author: I would go with option 1.
Unfortunately I don't have access to my laptop right now.

> * Some PDFs may contain a mix of scanned pages and pages with embedded text.

Reviewer: ```suggestion
* Some PDFs may contain a mix of pages with and without embedded text.
```
What is a scanned page?

Author: A page that is scanned using mobile for example, the one that is not digital born
Anyway I think your suggestion is better

> * Some PDFs may contain a mix of pages with and without embedded text.

Reviewer 2: I'm not sure if `*` would look good

Reviewer 3: Rich diff on github (although it will look different in jekyll)
<img width="1088" height="380" alt="image" src="https://github.com/user-attachments/assets/3f069a96-5628-4ce8-8f9f-e9d71760a4cc" />

Reviewer 4: We will see how it looks in gitbook

* In such cases, you will have three options:

> 1. You can choose to perform OCR only on the scanned pages(Skip pages with text), this is the default behavior if not specified.

Reviewer: ```suggestion
1. **Skip pages with text**: Performs OCR only on the pages without already embedded text. This is the default behaviour, if not specified.
```

2. Force OCR on all pages(Overwrite text in pages containing text).

> 3. Redo OCR for pages containing OCR text(Redo text in pages containing OCRed text).

Reviewer: ```suggestion
2. **Redo text in pages containing OCRed text**: Uses OCR on pages containing text created by a prior OCR run.
3. **Overwrite text in pages containing text**: Forces OCR with rasterization on all pages, potentially reducing quality or losing vector content, but this technique works, even when the `Redo text in pages containing text` option doesn't work.
```
